packages/oidc-client/src/lib/client.store.ts

```diff
@@ -180,5 +180,152 @@ export async function oidc<ActionType extends ActionTypes = ActionTypes>({
         }
       },
     },
+    user: {
+      info: async () => {
+        const state = store.getState();
+        const wellknown = wellknownSelector(wellknownUrl, state);
+
+        if (!wellknown?.userinfo_endpoint) {
+          const err = {
+            error: 'Wellknown missing userinfo endpoint',
+            type: 'wellknown_error',
+          } as AuthorizeErrorResponse;
+
+          log.error(err.error);
+
+          return err;
+        }
+
+        const tokens = (await storageClient.get()) as Record<string, string>;
+
+        if (!tokens || !('access_token' in tokens)) {
+          const err = {
+            error: 'No access token found',
+            type: 'auth_error',
+          } as AuthorizeErrorResponse;
+
+          log.error(err.error);
+
+          return err;
+        }
+
+        return await store.dispatch(
+          oidcApi.endpoints.userInfo.initiate({
+            accessToken: tokens.access_token,
+            endpoint: wellknown.userinfo_endpoint,
+          }),
+        );
+      },
+      logout: async () => {
+        const state = store.getState();
+        const wellknown = wellknownSelector(wellknownUrl, state);
+
+        if (!wellknown?.end_session_endpoint) {
+          const err = {
+            error: 'Wellknown missing end session endpoint',
+            type: 'wellknown_error',
+          } as AuthorizeErrorResponse;
+
+          log.error(err.error);
+
+          return err;
+        }
+
+        const tokens = (await storageClient.get()) as Record<string, string>;
+
+        if (!tokens || !('access_token' in tokens)) {
+          const err = {
+            error: 'No access token',
+            message: 'No access token found in storage; required for revoking access token',
+            type: 'state_error',
+          } as GenericError;
+
+          log.error(err.error);
+
+          return err;
+        }
+
+        if (!tokens || !('id_token' in tokens)) {
+          const err = {
+            error: 'No ID token',
+            message: 'No ID token found in storage; required for ending session',
+            type: 'state_error',
+          } as GenericError;
```

Comment:
again non blocking but couldn't we just type `err: GenericError` ? instead of needing to `as` cast it?